### PR TITLE
Finalize learner storage notifications

### LINKED ISSUE #10184: Finalize learner storage notifications
This is a followup to https://github.com/learningequality/kolibri/pull/10056 which was merged to get user strings merged before string freeze.

```[tasklist]
### Tasks
- [x] Make polling reactive to user logged in state, removing early return in poll function ([link](https://github.com/learningequality/kolibri/pull/10056/files#diff-f0d20dbe19423bd97d7aa86e34f39ba122e397094e9aabb714759e95ba124ad9R43))
- [x] ~Wire up `StorageNotification.vue` with appropriate data to show proper variations of notification message ([link](https://github.com/learningequality/kolibri/pull/10056/files#diff-155b7b2af6eb8f44179578e0b3670d02f3938df8170a9a2bcdaea7ff196dbcb7R70))~ **Replaced with** https://github.com/learningequality/kolibri/issues/10378
- [x] Properly handle cross-plugin routing ([link](https://github.com/learningequality/kolibri/pull/10056/files#r1094627926))
```

### PR DESCRIPTION
**Pull request by AllanOXDi**
## Summary

- Properly handle cross-plugin routing 
- Make polling reactive to user logged in state
closes #10184
## References
#10184
## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/composables/useUserSyncStatus.js
Comment (marcellamaki) [inline kolibri/core/assets/src/composables/useUserSyncStatus.js]: I'm not sure where you are ultimately using this value, but we might be able to use the 
`isUserLoggedIn` [value](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/state/modules/session.js#L36) from `session.js` instead

File: kolibri/core/assets/src/views/StorageNotification.vue
Comment (marcellamaki) [inline kolibri/core/assets/src/views/StorageNotification.vue]: Nice! This is a great and straightforward way to manage this that I hadn't thought about 👍

**Review by marcellamaki — COMMENTED** [review]
Nice work @AllanOXDi! I had a question about your changes in the useUserSyncStatus composable - I'm just trying to understand where these changes - especially the new `userLoggedIn` value are going to be used. After that I'll re-review!

File: kolibri/core/assets/src/composables/useUserSyncStatus.js
Comment (rtibbles) [reply kolibri/core/assets/src/composables/useUserSyncStatus.js]: Note there is also a composable for this too, which would be better used wherever this is being used (and can also be used in here where needed): https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/composables/useUser.js

File: kolibri/core/assets/src/views/StorageNotification.vue
Comment (rtibbles) [reply kolibri/core/assets/src/views/StorageNotification.vue]: There are a couple of potential issues here, the first is purely conceptual but more of a guide:

Core components should not have knowledge about plugins - this is encoding knowledge about a specific plugin into the core Kolibri code.

More practically - this will break of the device plugin has been disabled. Not only will it not redirect anywhere, it will also produce an error because `urls['kolibri:kolibri.plugins.device:device_management']` will be `undefined` so you'll have `TypeError: undefined is not a function` thrown.

The simple thing here to do is an existence check on the property before invoking it - the more systematic approach that addresses the first concern, I would need to fully understand what the intended behaviour is here and why the redirect to `/` was insufficient.

Comment (rtibbles) [reply kolibri/core/assets/src/views/StorageNotification.vue]: I just double checked, and it seems like the `/` redirect is a problem because of the redirect behaviour for learner only devices specifically.

I would suggest just doing the existence check here before invoking the function to avoid any potential errors and we can think about a more systematic approach in the future.

Comment (marcellamaki) [reply kolibri/core/assets/src/views/StorageNotification.vue]: Good point, @rtibbles ! I think the original `this.$router.push('/')` might work. The fact that this was a to-do was a lingering item that I had thought about several months ago, to make sure the redirecting was sufficient. But, since there have been changes in the redirect in general, the current option might work? 

There are two scenarios with notifications that require a redirect, and this would be for the superadmin case. The notification could appear at the top of any plugin, and the user would need to navigate to the device/channels page, according the the figma, to manage their content. 

Screenshot 2023-06-06 at 2 40 55 PM

File: kolibri/core/assets/src/views/StorageNotification.vue
Comment (marcellamaki) [inline kolibri/core/assets/src/views/StorageNotification.vue]: Hi @AllanOXDi - there are a couple more things here that I talked about with @rtibbles.

1. Using `canManageContent` is a more specific permission than checking that the user is not a learner, so you can use this (an import the getter that goes along with it)
2. Adding a check to make sure the URL exists would help with some of the routing concerns that Richard mentioned previously

```
 if (this.canManageContent && urls['kolibri:kolibri.plugins.device:device_management'])
```

Let us know if you have further questions

**Review by marcellamaki — CHANGES_REQUESTED** [review]
Hi @AllanOXDi - one more small tweak and then should be good to go!

File: kolibri/core/assets/src/views/StorageNotification.vue
Comment (AllanOXDi) [reply kolibri/core/assets/src/views/StorageNotification.vue]: Hi @marcellamaki  I think I have already  implemented  `canManageContent` in this[ PR ](https://github.com/learningequality/kolibri/pull/10709/files#:~:text=...mapGetters(%5B%27isLearner%27%2C%20%27isAdmin%27%2C%20%27canManageContent%27%5D)%2C)for permission checking.  I am assuming that this PR is following it?

Comment (AllanOXDi) [reply kolibri/core/assets/src/views/StorageNotification.vue]: I will first wait for the PR to get merge and get to this.

Comment (akolson) [reply kolibri/core/assets/src/views/StorageNotification.vue]: Hi @marcellamaki! I rebased this with develop after the merge of  #10709. I also added the check as advised in your comment above and some minor code streamlining to clarify the use of `showBanner` vs `bannerOpened` as the former was directly being set and yet was a computed property.

**Review by marcellamaki — APPROVED** [review]
Thanks for the teamwork here @AllanOXDi and @akolson! Let's get this merged, and as we get into manual QA with the full storage notification functionality, we can squash any bugs that pop up :)

**Pull request merged**
